crates/transpiler/src/equivalence.rs

```diff
@@ -709,46 +713,50 @@ impl EquivalenceLibrary {
     }
 }
 
-fn raise_if_param_mismatch(
-    py: Python,
+fn raise_if_param_mismatch<'a>(
     gate_params: &[Param],
-    circuit_parameters: Bound<PySet>,
+    circuit_parameters: impl Iterator<Item = &'a Symbol>,
 ) -> PyResult<()> {
-    let gate_params_obj = PySet::new(
-        py,
-        gate_params
-            .iter()
-            .filter(|param| matches!(param, Param::ParameterExpression(_))),
-    )?;
-    if !gate_params_obj.eq(&circuit_parameters)? {
+    let parsed_gate_params: HashSet<Symbol> = gate_params
+        .iter()
+        .filter_map(|param| match param {
+            Param::ParameterExpression(parameter_expression) => {
+                parameter_expression.try_to_symbol().ok()
+            }
+            _ => None,
+        })
+        .collect();
+    let circuit_parameters: HashSet<Symbol> = circuit_parameters.cloned().collect();
+    if parsed_gate_params != circuit_parameters {
```

Comment:
Do you need a second allocation here, can you just do?:

```suggestion
    if !circuit_parameters.any(|x| parsed_gate_params.contains(x)) {
        let circuit_parameters: HashSet<Symbol> = circuit_parameters.cloned().collect();
```

Reply:
Yes! I would also have to check for the sizes before doing this but it works!

Reply:
Corrected in 5cf3db5ddb1b7fff29a21f8ddb462e206f6d7554